cpp/src/strings/convert/convert_ipv4.cu

```diff
@@ -204,6 +205,49 @@ std::unique_ptr<column> integers_to_ipv4(
                              mr);
 }
 
+std::unique_ptr<column> is_ipv4(strings_column_view const& strings,
+                                cudaStream_t stream,
+                                rmm::mr::device_memory_resource* mr)
+{
+  auto strings_column = column_device_view::create(strings.parent(), stream);
+  auto d_column       = *strings_column;
+  // create output column
+  auto results   = make_numeric_column(data_type{type_id::BOOL8},
+                                     strings.size(),
+                                     copy_bitmask(strings.parent(), stream, mr),
+                                     strings.null_count(),
+                                     stream,
+                                     mr);
+  auto d_results = results->mutable_view().data<bool>();
+  thrust::transform(rmm::exec_policy(stream)->on(stream),
+                    thrust::make_counting_iterator<size_type>(0),
+                    thrust::make_counting_iterator<size_type>(strings.size()),
+                    d_results,
+                    [d_column] __device__(size_type idx) {
+                      if (d_column.is_null(idx)) return false;
+                      auto const d_str = d_column.element<string_view>(idx);
+                      if (d_str.empty()) return false;
+                      constexpr int max_ip = 255;  // values must be in [0,255]
+                      int ip_vals[4]       = {-1, -1, -1, -1};
+                      int ipv_idx          = 0;
+                      for (auto const ch : d_str) {
+                        if ((ch >= '0') && (ch <= '9')) {
+                          auto const ip_val    = ip_vals[ipv_idx];
+                          int const new_ip_val = static_cast<int>(ch - '0') +  // compute new value
+                                                 (ip_val < 0 ? 0 : (10 * ip_val));
+                          if (new_ip_val > max_ip) return false;
+                          ip_vals[ipv_idx] = new_ip_val;
+                        } else if (ch != '.' || (++ipv_idx > 3))
```

Comment:
The code that increments `ipv_idx` is a bit tricky, might benefit from a comment.